lib/seattleflu/id3c/cli/command/etl/presence_absence.py

```diff
@@ -0,0 +1,449 @@
+"""
+Process presence-absence tests that are specific to the custom JSON dump
+from Samplify to create standardized FHIR bundles that are inserted into
+the receiving FHIR table.
+
+Each FHIR bundle will contain results for one sample so that if errors arise,
+results for each sample can get processed separately without blocking ingestion
+of results of other samples.
+
+The presence-absence ETL process will abort under these conditions:
+
+1. If we receive an unexpected value for the top-level key
+
+2. If we receive a bogus "chip" value
+
+3. If we receive an unexpected value for the "controlStatus" of a target
+
+4. If we receive an unexpected value for the "targetResult" of a specific test
+
+5. If we receive an unexpected value for the "assayName"
+
+6. If we receive an unexpected value for the "assayType"
```

Comment:
Worth noting that we have 6 conditions where the ETL will abort. These don't happen that often and seem to be pretty major things that we would immediately want to address. It would be great to hear others' opinions on whether any of these checks should be loosened to just skip with warnings instead of stopping the entire pipeline.